### PR TITLE
Use mexify, change C++ forwardKin, centerOfMass method signatures, add dynamicsBias

### PR DESCRIPTION
**Pull request by tkoolen**
* introduce mexify, which makes it easy to create and maintain mex functions given a C++ function
 * compiles on MSVC now, worked around MSVC internal compiler error
 * talked to @rdeits; Swig is still the long term solution for this, but I still need something now to reduce my overhead and increase my sanity as I change method signatures and get rid of GradientVar in favor of AutoDiffScalar stuff soon.
* convert a bunch of mex functions over to using mexify
* change forwardKin and centerOfMass so that they do *not* return GradientVars anymore, just matrices (like everything else, in the near future). This is a first step in the direction of AutoDiffScalar gradients. Note that forwardKin and centerOfMass gradients (Jacobians) are used in online control code and hence continue to require associated dedicated Jacobian methods that do not require an AutoDiffScalar KinematicsCache, which would be too costly to compute.
* add dynamicsBias method as agreed upon during triage meeting; also made the method signature of inverseDynamics a bit nicer to work with.

### CODE REVIEW

**Comment by tkoolen** [comment]
Bug report for MSVC internal compiler error I had to work around:
https://connect.microsoft.com/VisualStudio/feedback/details/1847159

**Comment by tkoolen** [comment]
Compile error on drake003 is because mxIsScalar was only introduced in Matlab 2015a.

While investigating this, I noticed that drake003 is set up to use Matlab 2012a, while using Visual Studio 2013. VS 2013 isn't even supported on Matlab 2012a.

**Comment by tkoolen** [comment]
Snopt runs into the major iteration limit testing QuasiStaticConstraint in testIK. I can reproduce this on my machine. The test passes if I increase the major iteration limit from 3000 to 5000. However the second time I try to run it, the test passes much more quickly (and doesn't use as many major iterations).

@hongkai-dai, @avalenzu, do you think I did something wrong (I did touch related code), or is this just a snoptism?

**Comment by avalenzu** [comment]
@tkoolen, it seems odd that that would change. Is there a way to test the constraint and gradient values before and after your changes?

**Comment by tkoolen** [comment]
@avalenzu, I'll check.

**Comment by tkoolen** [comment]
Actually, systems/plants/constraint/test/testQuasiStaticConstraint checks constraint values and gradients against the Matlab implementation which hasn't changed, and passes just fine.

File: drake/systems/plants/RigidBodyManipulator.cpp [hunk 1]
```diff
@@ -642,7 +640,7 @@ GradientVar<Scalar, TWIST_SIZE, Eigen::Dynamic> RigidBodyManipulator::centroidal
   auto ret = worldMomentumMatrix(cache, gradient_order, robotnum, in_terms_of_qdot);
 
   // transform from world frame to COM frame
-  Matrix<Scalar, SPACE_DIMENSION, 1> com = centerOfMass(cache, 0).value();
+  Matrix<Scalar, SPACE_DIMENSION, 1> com = centerOfMass(cache, robotnum);
```
Comment (tkoolen) [inline drake/systems/plants/RigidBodyManipulator.cpp]: This was actually a bad bug.

**Comment by tkoolen** [comment]
I think the QuasiStaticConstraint issue may just be random seed related.
* start matlab, addpath_drake; cd systems/plants/test; testIK --> Check quasi static constraint takes a long time
* start matlab, addpath_drake; rng(1, 'twister'); cd systems/plants/test; testIK --> Check quasi static constraint finishes quickly

File: drake/systems/plants/@RigidBodyManipulator/findKinematicPath.m [hunk 2]
```diff
@@ -20,7 +20,10 @@
 end
 
 if use_mex
-  [body_path, joint_path, signs] = findKinematicPathmex(obj.mex_model_ptr, start_body, end_body);
+  [body_path, joint_path, signs] = findKinematicPathmex(obj.mex_model_ptr, start_body - 1, end_body - 1); % base 1 to base 0
```
Comment (psiorx) [inline drake/systems/plants/@RigidBodyManipulator/findKinematicPath.m]: I thought the convention was to have index conversions happen in the mex file.

Comment (tkoolen) [inline drake/systems/plants/@RigidBodyManipulator/findKinematicPath.m]: That can't be maintained in general in this kind of framework. One C++ argument type needs to map to one conversion method from Matlab, at least within one function. Right now we have e.g.
```
  template <typename Scalar>
  GradientVar<Scalar, TWIST_SIZE, 1> geometricJacobianDotTimesV(const KinematicsCache<Scalar>& cache, int base_body_or_frame_ind, int end_effector_body_or_frame_ind, int expressed_in_body_or_frame_ind, int gradient_order) const;
```
where `base_body_or_frame_ind`, `end_effector_body_or_frame_ind`, `expressed_in_body_or_frame_ind`, and `gradient_order` are all ints, but `gradient_order` should not have a base 1 to base 0 conversion applied, while the others do.

I think the long term solution is to actually have these arguments be of different types, i.e. turn the `*_body_or_frame_ind` arguments into `const RigidBodyFrame&`. In fact, in this case `gradient_order` is going to disappear when we switch over to AutoDiffScalar gradients.

**Comment by tkoolen** [comment]
Any more comments? Ready to merge?

**Pull request merged**